blocks/procedures.js

```diff
@@ -980,7 +980,7 @@ const PROCEDURE_CALL_COMMON = {
         Xml.domToWorkspace(xml, this.workspace);
         Events.setGroup(false);
       }
-    } else if (event.type === Events.BLOCK_DELETE) {
+    } else if (event.type === Events.BLOCK_DELETE && event.blockId != this.id) {
```

Comment:
Strictly speaking this is a breaking change and should probably be mentioned in the PR description, even if it's very unlikely to ever cause anyone an issue.

Reply:
I don't think this is a breaking change. Before blocks did not receive their own events, so this case would have never been hit. Now I've added the `blockId` check so it continues to not be hit.

Reply:
`onchange` is not private, so it could be called by code outside Blockly, which could pass in an arbitrary event.

I agree this is unlikely in practice, which is why I said "strictly speaking".  It's probably not useful to bump a major version number or to mention in the release notes, but it probably _is_ worth making sure that I am not the only one who has considered this scenario, and hence the comment.